Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -721,5 +739,105 @@ GStreamerRegistryScanner::RegistryLookupResult GStreamerRegistryScanner::isConfi
+
+    if (factories.hasElementForMediaType(codecElement, "audio/isac") && factories.hasElementForMediaType(rtpElement, "audio/isac"))
+        capabilities.codecs.append({ .mimeType = "audio/ISAC"_s, .clockRate = 16000, .channels = 1, .sdpFmtpLine = emptyString() });
+
+    if (factories.hasElementForMediaType(codecElement, "audio/G722") && factories.hasElementForMediaType(rtpElement, "audio/G722"))
+        capabilities.codecs.append({ .mimeType = "audio/G722"_s, .clockRate = 8000, .channels = 1, .sdpFmtpLine = emptyString() });
+
+    if (factories.hasElementForMediaType(codecElement, "audio/x-mulaw") && factories.hasElementForMediaType(rtpElement, "audio/x-mulaw"))
+        capabilities.codecs.append({ .mimeType = "audio/PCMU"_s, .clockRate = 8000, .channels = 1, .sdpFmtpLine = emptyString() });
+
+    if (factories.hasElementForMediaType(codecElement, "audio/x-alaw") && factories.hasElementForMediaType(rtpElement, "audio/x-alaw"))
+        capabilities.codecs.append({ .mimeType = "audio/PCMA"_s, .clockRate = 8000, .channels = 1, .sdpFmtpLine = emptyString() });
+}
+
+void GStreamerRegistryScanner::fillVideoRtpCapabilities(Configuration configuration, RTCRtpCapabilities& capabilities)
+{
+    if (!m_videoRtpExtensions)
+        m_videoRtpExtensions = probeRtpExtensions(m_allVideoRtpExtensions);
+    if (m_videoRtpExtensions)
+        capabilities.headerExtensions = copyToVector(*m_videoRtpExtensions);
+
+    auto codecElement = ElementFactories::Type::VideoDecoder;
+    auto rtpElement = ElementFactories::Type::RtpDepayloader;
+    if (configuration == Configuration::Encoding) {
+        codecElement = ElementFactories::Type::VideoEncoder;
+        rtpElement = ElementFactories::Type::RtpPayloader;
+    }
+
+    auto factories = ElementFactories({ codecElement, rtpElement });
+
+    if (factories.hasElementForMediaType(codecElement, "video/x-h264") && factories.hasElementForMediaType(rtpElement, "video/x-h264")) {
+        // Profile levels are hardcoded here for the time being. It might be a good idea to actually
+        // probe those on the selected encoder.
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=640c1f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42e01f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=640c1f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=42e01f"_s });
+    }
+
+    // TODO: Probe for video/H265 capabilies.
+    // TODO: Probe for video/AV1 capabilies.
```

Comment:
IIRC, the styles we can only have FIXMEs.

Reply:
FIXME it is then